src/scenes/CreateEvent/store.js

```diff
@@ -252,8 +252,20 @@ export default class CreateEventStore {
       return;
     }
 
-    if (await this.hasPendingCreateTxs()) return;
-    if (await !this.getEscrowAmount()) return;
+    // Close if getting pending txs fails
+    const hasPendingTxs = await this.hasPendingCreateTxs();
+    if (hasPendingTxs) {
+      this.close();
+      return;
+    }
+
+    // Close if unable to get the escrow amount
+    const escrowAmountSuccess = await this.getEscrowAmount();
```

Comment:
By playing around, I notice that
```js
if (!(await this.getEscrowAmount())) return;
```
will achieve the same. Previously we had `!` before `this.getEscrowAmount`, which did not work.

Why it did not work? I'm guessing it's because `await` was waiting for `!` not `this.getEscrowAmount`. But I'm doubting, do you know the exact reason?

Reply:
Ah ya you are right, adding the not operator with the await enclosed in brackets will do the same `(!(await this.getEscrowAmount()))`.

This does not work:
`if (await !this.getEscrowAmount()) return;`
because `this.getEscrowAmount()` returns a Promise so `awaiting` it returns the proper `boolean` return.

Reply:
oh ok, but what does `!` catch if `this.getEscrowAmount()` not finished? Does it just catch `true` since it did not reach `return;`?

Reply:
You don't need to change if you think both are equal.